spi/common/core-spi/src/main/java/org/eclipse/edc/spi/types/domain/secret/Secret.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2024 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.edc.spi.types.domain.secret;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.edc.spi.entity.Entity;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.eclipse.edc.spi.constants.CoreConstants.EDC_NAMESPACE;
+
+
+/**
+ * The {@link Secret} contains the metadata and describes the data itself or a collection of data.
+ */
+@JsonDeserialize(builder = Secret.Builder.class)
+public class Secret extends Entity {
+    public static final String PROPERTY_ID = EDC_NAMESPACE + "id";
+    public static final String EDC_SECRET_TYPE = EDC_NAMESPACE + "Secret";
+
+    public static final String EDC_SECRET_VALUE = EDC_NAMESPACE + "value";
+
+    private String value;
+
+
+    private Secret() {
+    }
+
+
+    public String getValue() {
+        return value;
+    }
+
+    public Builder toBuilder() {
+        // No key, Id is used as vault key
+        return Builder.newInstance()
+                .value(value);
+    }
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static class Builder extends Entity.Builder<Secret, Builder> {
```

Comment:
do we expect this to be extensible? if not, i would consider making the constructor private, and possibly making the Builder class `final`